src/views/__tests__/CTLViewsHoc.useListingHocInt.js

```diff
@@ -0,0 +1,430 @@
+	// call a navigate function. In these examples it is expected that the
+	// wrapped component would actually, and correctly, be called twice.
+	// These tests should not trigger a redirect based on ID.
+
+	// TODO: Add a test for one ID to another.
+	it('should handle the request for a concept, and then handle a navigate to another concept', async () => {
+		getListingInformationByName.mockReturnValueOnce(CONCEPT_BREAST_CANCER);
+
+		getListingInformationById.mockReturnValueOnce(CONCEPT_NO_PURL);
+
+		// We use RedirectPath to figure out the correct route for
+		// a redirect. This assumes /:codeOrPurl which matches our
+		// route below.
+		const mockRedirectPath = jest.fn().mockReturnValue('/C99999');
+
+		// This component will force a navigation
+		const mockComponent = jest.fn(({ data }) => {
+			const navigate = useNavigate();
+			useEffect(() => {
+				// Only navigate if the concept is our first one.
+				if (data.some((concept) => concept.conceptId.includes('C4872'))) {
+					navigate('/C99999', {
+						replace: true,
+					});
+				}
+			}, []);
+			return <></>;
+		});
+
+		const WrappedComponent = CTLViewsHoC(mockComponent);
+
+		// Hopefully this renders, and navigates and loads the
+		// new component.
+		await act(async () => {
+			render(
+				<MockAnalyticsProvider>
+					<MemoryRouter initialEntries={['/breast-cancer']}>
+						<Routes>
+							<Route
+								path="/:codeOrPurl"
+								element={
+									<WrappedComponent
+										redirectPath={mockRedirectPath}
+										routeParamMap={diseaseRouteParamMap}
+									/>
+								}
+							/>
+						</Routes>
+					</MemoryRouter>
+				</MockAnalyticsProvider>
+			);
+		});
+
+		// First request
+		expect(getListingInformationByName.mock.calls).toHaveLength(1);
+		expect(getListingInformationByName.mock.calls[0][1]).toEqual(
+			'breast-cancer'
+		);
+
+		// Call after redirect
+		expect(getListingInformationById.mock.calls).toHaveLength(1);
+		expect(getListingInformationById.mock.calls[0][1]).toEqual(['C99999']);
+
+		// This should have only 2 calls. One for the Purl, and one
+		// for the C-code without the purl.
+		expect(mockComponent.mock.calls).toHaveLength(2);
+		const { data: call1data } = mockComponent.mock.calls[0][0];
+		const { data: call2data } = mockComponent.mock.calls[1][0];
+
+		expect(call1data).toEqual([CONCEPT_BREAST_CANCER]);
+
+		expect(call2data).toEqual([CONCEPT_NO_PURL]);
+	});
+
+	// TODO: Add a test for route with 1 param to a route with 2 params
+	it('should handle the request for a concept, and then handle a navigate to concept plus type', async () => {
+		getListingInformationById.mockImplementation((client, ids) => {
+			if (ids.includes('C99999')) {
+				return CONCEPT_NO_PURL;
+			}
+		});
+
+		getTrialType.mockImplementation((client, trialType) => {
+			if (trialType === 'treatment') {
+				return TRIAL_TYPE_TREATMENT;
+			}
+		});
+
+		// This should not get called.
+		const mockRedirectPath = jest.fn().mockReturnValue('');
```

Comment:
Same as above

Reply:
Fair point, but I will meet you half-way. I will just have the mockImplementation throw and that way the code blows up cause it should not get called at all.

Right now for this PR without caching, if the redirect had occurred then the count for the API calls would be off.

Reply:
👍